hledger-lib/Hledger/Data/Dates.hs

```diff
@@ -736,31 +854,62 @@ reportinginterval = choice' [
                           return $ Months 2,
                        do string "every"
                           many spacenonewline
-                          n <- fmap read $ some digitChar
-                          thsuffix
+                          n <- nth
                           many spacenonewline
                           string "day"
                           many spacenonewline
-                          string "of"
+                          of_ "week"
+                          return $ DayOfWeek n,
+                       do string "every"
+                          many spacenonewline
+                          n <- weekday
                           many spacenonewline
-                          string "week"
                           return $ DayOfWeek n,
                        do string "every"
                           many spacenonewline
-                          n <- fmap read $ some digitChar
-                          thsuffix
+                          n <- nth
                           many spacenonewline
                           string "day"
-                          optional $ do
-                            many spacenonewline
-                            string "of"
-                            many spacenonewline
-                            string "month"
-                          return $ DayOfMonth n
+                          many spacenonewline
```

Comment:
Why you added this `many spacenonewline` as mandatory? It was under `optional` and I gues it is already part of `of_`?

Reply:
Without it, existing parser misbehaved:
```
*Hledger.Data.Dates> p $ T.pack "every 2nd day 2009"
Right (NoInterval,DateSpan -)
*Hledger.Data.Dates> p $ T.pack "every 2nd day 2009-"
Right (NoInterval,DateSpan -)
*Hledger.Data.Dates> p $ T.pack "every 2nd day from 2009-"
Right (NoInterval,DateSpan -)
```

Reply:
This means that problem is somewhere else.

You can catch more errors with adding `eof` to the end of the parser
```haskell
-- >>> let p s = parsewith (periodexpr (parsedate "2008/11/26") <* eof) (T.toLower s) :: Either (ParseError Char MPErr) (Interval, DateSpan)
```

And error is in use of `optional`. It look slike we miss `try`. We dive into `optional` fail on `of` and return success, but consume spaces.

Reply:
Example of parser failure from this change:
```sh
% hledger reg -p 'every tue of week' 
hledger: could not parse period option: 1:11:
unexpected 'o'
expecting end of input
 (use -h to see usage)
```